MapboxSearch/sdk/src/main/java/com/mapbox/search/result/SearchAddress.kt

```diff
@@ -87,7 +87,7 @@ public class SearchAddress @JvmOverloads public constructor(
 
         return when {
             fields.isEmpty() -> null
-            components.first() == HOUSE_NUMBER -> {
+            components.first() == HOUSE_NUMBER && !houseNumber.isNullOrEmpty() -> {
```

Comment:
Can `HOUSE_NUMBER` only be the first component? this seems fragile and I would expect adding ` ` around house number based on nearby separators. Can you rethink this from scratch, making sure same set of tests passes?

Reply:
Discussed internally and decided to remove `FormatStyle.Custom`. All the other `FormatStyle`'s start with the `HOUSE_NUMBER` so existing logic already works. `FormatStyle.Custom` makes no much sense given that `SearchAddress` provides all the address components in the public api and customers have possibility to format them with any needed rules.